pymobiledevice3/services/mobile_image_mounter.py

```diff
@@ -338,23 +340,31 @@ def auto_mount_developer(
     image_mounter.mount(image_path, signature)
 
 
-async def auto_mount_personalized(lockdown: LockdownServiceProvider) -> None:
+async def auto_mount_personalized(lockdown: LockdownServiceProvider, version: Optional[str] = None) -> None:
     local_path = get_home_folder() / 'Xcode_iOS_DDI_Personalized'
     local_path.mkdir(parents=True, exist_ok=True)
 
     image = local_path / 'Image.dmg'
     build_manifest = local_path / 'BuildManifest.plist'
     trustcache = local_path / 'Image.trustcache'
-
+    ddi_id = (LATEST_DDI_BUILD_ID if version is None else version);
     if (not build_manifest.exists() or
-            plistlib.loads(build_manifest.read_bytes()).get('ProductBuildVersion') != LATEST_DDI_BUILD_ID):
+            plistlib.loads(build_manifest.read_bytes()).get('ProductBuildVersion') != ddi_id):
         # download the Personalized image from our repository
-        repo = DeveloperDiskImageRepository.create()
-        personalized_image = repo.get_personalized_disk_image()
-
-        image.write_bytes(personalized_image.image)
-        build_manifest.write_bytes(personalized_image.build_manifest)
-        trustcache.write_bytes(personalized_image.trustcache)
+        try:
+            repo = DeveloperDiskImageRepository.create()
+            personalized_image = repo.get_personalized_disk_image()
+
+            image.write_bytes(personalized_image.image)
+            build_manifest.write_bytes(personalized_image.build_manifest)
+            trustcache.write_bytes(personalized_image.trustcache)
+        except:
+            if not build_manifest.exists():
+                raise;
+            else:
+                logger.debug("Use local DeveloperDiskImage(ProductBuildVersion= %s)" % (plistlib.loads(build_manifest.read_bytes()).get("ProductBuildVersion")));
```

Comment:
- No need for the `else` statement as any other condiftion raises an exception
- This sound like an actual error that should be raised. I see no point for this error message